nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/StandardRemoteGroupPort.java

```diff
@@ -366,6 +371,14 @@ private int receiveFlowFiles(final Transaction transaction, final ProcessContext
         return flowFilesReceived.size();
     }
 
+    private String createTransitUri(Transaction transaction, String sourceFlowFileIdentifier) {
+        if (transaction instanceof CustomTransitUriTransaction) {
```

Comment:
@markap14 Thanks for the comment, I agree with that. I felt that I should have kept Peer completely immutable, so I might bypass the createTransitUri call to CommunicationsSession. Will update the PR soon.